### PR TITLE
Feature/gp creation

### PR DESCRIPTION
**Pull request by kenjao**
- Created GP Endpoints
- Created GP Service
- Created GP Repository
- Created GP Model that extends User
- Hashed GP Password

### CODE REVIEW

File: src/main/java/ie/ait/bteam/drcare/data/model/GeneralPractitioner.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ie.ait.bteam.drcare.data.model;
+
+import javax.persistence.Entity;
+
+@Entity
+public class GeneralPractitioner extends User {
```
Comment (adheli) [inline src/main/java/ie/ait/bteam/drcare/data/model/GeneralPractitioner.java:6]: I don't see the need for this new entity, as all the GP attributes are still in User. Better remove it, as UserValidation can check the GP values and when adding security, we don't have to check diff entities.

File: src/main/java/ie/ait/bteam/drcare/data/model/User.java [hunk 2]
```diff
@@ -36,13 +36,14 @@
 	@Getter
 	@Setter
 	@Column(unique = true)
+	@NotNull
```
Comment (adheli) [inline src/main/java/ie/ait/bteam/drcare/data/model/User.java]: Patient doesn't need to have email, so remove this pls.

File: src/main/java/ie/ait/bteam/drcare/data/model/User.java [hunk 3]
```diff
@@ -36,13 +36,14 @@
...
 	private String email;
 
 	@Getter
 	@Setter
 	@Column
 	@NotNull
-	private Boolean isAdmin;
+	private Boolean isAdmin = false;
```
Comment (adheli) [inline src/main/java/ie/ait/bteam/drcare/data/model/User.java]: boolean by default is false. if you need to do any of this, do on DTOs

File: src/main/java/ie/ait/bteam/drcare/DrCareApplication.java [hunk 4]
```diff
@@ -3,18 +3,21 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.web.servlet.ServletComponentScan;
 import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
 
 @SpringBootApplication
-public class DrCareApplication extends SpringBootServletInitializer {
```
Comment (adheli) [inline src/main/java/ie/ait/bteam/drcare/DrCareApplication.java:9]: please put extends back to expose the controllers to outside spring context.

File: src/main/java/ie/ait/bteam/drcare/DrCareApplication.java [hunk 5]
```diff
@@ -3,18 +3,21 @@
...
+@ServletComponentScan
+public class DrCareApplication{
+//public class DrCareApplication extends SpringBootServletInitializer {
 
 	public static void main(String[] args) {
 		SpringApplication.run(DrCareApplication.class, args);
 	}
 
-	@Override
```
Comment (adheli) [inline src/main/java/ie/ait/bteam/drcare/DrCareApplication.java]: after adding back the extends, uncomment this.

File: src/main/java/ie/ait/bteam/drcare/config/SecurityConfiguration.java [hunk 6]
```diff
@@ -1,15 +1,24 @@
...
+	@Bean
+	public BCryptPasswordEncoder bCryptPasswordEncoder() {
+		return new BCryptPasswordEncoder();
+	}
+
 	@Override
 	protected void configure(HttpSecurity http) throws Exception {
 		// for now permission to all requests
-		http.authorizeRequests().anyRequest().permitAll();
+		//http.cors().and().csrf().disable();
```
Comment (adheli) [inline src/main/java/ie/ait/bteam/drcare/config/SecurityConfiguration.java]: remove commented code pls.

**Review by adheli — CHANGES_REQUESTED** [review]
Please review all my comments.

File: src/main/java/ie/ait/bteam/drcare/data/model/User.java (hunk 3 above)
Comment (kenjao) [reply src/main/java/ie/ait/bteam/drcare/data/model/User.java]: Object Boolean is null by default while primitive is false... That's why I initialised this to false. Will change it now tho.

File: src/main/java/ie/ait/bteam/drcare/data/model/User.java (hunk 2 above)
Comment (kenjao) [reply src/main/java/ie/ait/bteam/drcare/data/model/User.java]: okay done...

File: src/main/java/ie/ait/bteam/drcare/config/SecurityConfiguration.java (hunk 6 above)
Comment (kenjao) [reply src/main/java/ie/ait/bteam/drcare/config/SecurityConfiguration.java]: Removed

File: src/main/java/ie/ait/bteam/drcare/DrCareApplication.java (hunk 5 above)
Comment (kenjao) [reply src/main/java/ie/ait/bteam/drcare/DrCareApplication.java]: Uncommented

File: src/main/java/ie/ait/bteam/drcare/DrCareApplication.java (hunk 4 above)
Comment (kenjao) [reply src/main/java/ie/ait/bteam/drcare/DrCareApplication.java:9]: Done...

**Review by seun-odebunmi — APPROVED** [review]
Ok

**Pull request merged**